tools/harness-simulation/harness/Thread_Harness/THCI/OpenThread_Sim.py

```diff
@@ -0,0 +1,188 @@
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the
+#    names of its contributors may be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+"""
+>> Thread Host Controller Interface
+>> Device : OpenThread_Sim THCI
+>> Class : OpenThread_Sim
+"""
+
+import ipaddress
+import logging
+import paramiko
+import re
+import socket
+import time
+
+from IThci import IThci
+from OpenThread import OpenThreadTHCI, watched
+from simulation.config import REMOTE_OT_CLI_FTD_PATH
+
+
+logging.getLogger('paramiko').setLevel(logging.INFO)
+
+
+class SSHHandle(object):
+
+    def __init__(self, ip, port, username, password, device, node_id):
+        ipaddress.ip_address(ip)
+        self.ip = ip
+        self.port = int(port)
+        self.username = username
+        self.password = password
+        self.__handle = None
+        self.__channel = None
+        self.__connect(device, node_id)
+
+    @watched
+    def __connect(self, device, node_id):
+        self.close()
+
+        self.__handle = paramiko.SSHClient()
+        self.__handle.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        try:
+            logging.info('Connecting to %s:%s with username=%s', self.ip, self.port, self.username)
+            self.__handle.connect(self.ip, port=self.port, username=self.username, password=self.password)
+        except paramiko.ssh_exception.AuthenticationException:
+            if not self.password:
+                self.__handle.get_transport().auth_none(self.username)
+            else:
+                raise Exception('Password error')
+
+        self.__channel = self.__handle.get_transport().open_session()
+        self.__channel.invoke_shell()
+        self.__channel.setblocking(0)
+        
+        # Skip the login message
+        self.execute('echo Done', 10)
```

Comment:
All the commands but this one are executed in ot-cli-ftd rather than the shell. I intended to keep the `execute` method for further use when it comes to execute commands in the shell, but it seems unnecessary now. I will inline the method here then.

Reply:
I don't mean that execute is used once so that it should be inlined. 

I was asking why do we execute `echo Done` command here. What's the purpose of this executing this command?

Reply:
`echo Done` here is actually a hack trick to skip the login message. When using `invoke_shell()`, first output will be something like
```
The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
```
But I think this output will differ on different devices. I use this trick to skip the login message. An alternative solution is just simply waiting enough time to make sure the message has been printed completely.

---

By the way, I found an approach to interact with PTY without invoking a shell. It would not print any login messages after I tested. Is it necessary to change the implementation of SSH connection in this PR, or leave it in the next PR?